master/internal/api_trials.go

```diff
@@ -158,3 +158,52 @@ func (a *apiServer) KillTrial(
 	}
 	return &resp, err
 }
+
+func (a *apiServer) GetExperimentTrials(
+	_ context.Context, req *apiv1.GetExperimentTrialsRequest,
+) (*apiv1.GetExperimentTrialsResponse, error) {
+	resp := &apiv1.GetExperimentTrialsResponse{}
+
+	switch err := a.m.db.QueryProto(
+		"proto_get_trials_for_experiment",
+		&resp.Trials,
+		req.ExperimentId,
+	); {
+	case err == db.ErrNotFound:
+		return nil, status.Errorf(codes.NotFound, "experiment %d not found:", req.ExperimentId)
+	case err != nil:
+		return nil, err
+	}
+	a.filter(&resp.Trials, func(i int) bool {
+		v := resp.Trials[i]
+		// If there are no filters, return true.
```

Comment:
We probably don't need these comments, I just left them to explain why I suggested what I did.